templates/accounts/registration.html

```diff
@@ -1,47 +1,76 @@
 {% extends "accounts/_section.html" %}
-
+{% load util %}
 {% block title %}registration{% endblock %}
 
 {% block section_content %}
 <style> label[for=id_accepted_tos] {display: inline} #username_check, #check_success, #spinner { display:none; color: #bd2d33; font-style: italic; } #check_success {color:green;}</style>
 <script type="text/javascript">
     $(function () {
-      $("#id_username").focus()
-      
-        $('#id_username').parent().append('<div id="spinner"> Checking...<img src="{{ media_url }}images/indicator.gif" /></div><div id="username_check">Sorry, this username is already being used by someone else.</div><div id="check_success">Username is available</div>');
-      
-      // When the user enters a username we check if it's available
-      $("#id_username").focusout(function(){
-        if ($("#id_username").val()) {
-          $('#spinner').show();
-          $('#username_check').hide();
-          $('#check_success').hide();
-          $.get('{% url "check_username" %}', 
-              {'username': $("#id_username").val()}, 
-              function(data){
-                $('#username_check').toggle(!data.result);
-                $('#spinner').hide();
-                $('#check_success').toggle(data.result);
-              }, "json" );
+        $("#id_username").focus()
+            $('#id_username').parent().append('<div id="spinner"> Checking...<img src="{{ media_url }}images/indicator.gif" /></div><div id="username_check">Sorry, this username is already being used by someone else.</div><div id="check_success">Username is available</div>');
+            $("#id_username").focusout(function(){
+            if ($("#id_username").val()) {
+                $('#spinner').show();
+                $('#username_check').hide();
+                $('#check_success').hide();
+                $.get('{% url "check_username" %}', {'username': $("#id_username").val()}, function(data){
+                        $('#username_check').toggle(!data.result);
+                        $('#spinner').hide();
+                        $('#check_success').toggle(data.result);
+                }, "json" );
+            }
+        });
+    });
+
+    function togglePasswordInputType() {
+        var passwordInput = document.getElementById("id_{{ form.password1.name }}");
+        var eyeImg = document.getElementById("eye-img");
+        if (passwordInput.type === 'password'){
```

Comment:
We should probably make the submit form change this back to type=password if a user has set it, so that we can hint to the browser that it's a login and that it should offer to save it